src/Http/Http.Abstractions/src/Extensions/UseExtensions.cs

```diff
@@ -29,5 +38,22 @@ public static IApplicationBuilder Use(this IApplicationBuilder app, Func<HttpCon
                 };
             });
         }
+
+        /// <summary>
+        /// Adds a middleware delegate defined in-line to the application's request pipeline.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/> instance.</param>
+        /// <param name="middleware">A function that handles the request or calls the given next function.</param>
+        /// <returns>The <see cref="IApplicationBuilder"/> instance.</returns>
+        public static IApplicationBuilder Use(this IApplicationBuilder app, Func<HttpContext, RequestDelegate, Task> middleware)
+        {
+            return app.Use(next =>
+            {
+                return context =>
+                {
+                    return middleware(context, next);
+                };
+            });
```

Comment:
Make it a programming puzzle

```suggestion
            return app.Use(next => context => middleware(context, next));
```